cmd/root.go

```diff
@@ -253,3 +299,85 @@ func makeRepo(uri string) (clients.Repo, error) {
 
 	return nil, fmt.Errorf("unable to parse as github, gitlab, or azuredevops: %w", compositeErr)
 }
+
+// processRepo performs the scanning and formatting for a single repo URI.
+// It returns the Result when successful (or when combined output is requested),
+// or an error describing why the URI should be skipped.
+func processRepo(
+	ctx context.Context,
+	uri string,
+	o *options.Options,
+	enabledProbes []string,
+	enabledChecks checker.CheckNameToFnMap,
+	checksList []string,
+	opts []scorecard.Option,
+	checkDocs docs.Doc,
+	pol *policy.ScorecardPolicy,
+) (*scorecard.Result, error) {
+	var repo clients.Repo
+	var err error
+
+	if o.Local != "" && uri == o.Local {
+		repo, err = localdir.MakeLocalDirRepo(uri)
+		if err != nil {
+			return nil, fmt.Errorf("localdir: %w", err)
+		}
+	} else {
+		repo, err = makeRepo(uri)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	label := repoLabelFromURI(uri)
+
+	// Start banners with repo label (always show banners even in combined-only mode)
+	if o.Format == options.FormatDefault {
+		if len(enabledProbes) > 0 {
+			printProbeStart(label, enabledProbes)
+		} else {
+			printCheckStart(label, enabledChecks)
+		}
```

Comment:
let's get rid of `repoLabelFromURI` and just use `uri` instead of `label`

Otherwise it would be confusing when you analyze repos across forges:
```
--repos=github.com/some/repo,gitlab.com/other/repo
```

Which one is github? Which one is gitlab? What if they both have the exact same repo name?
```
Starting (some/repo) [Maintained]
...
Starting (other/repo) [Maintained]
```